### PR TITLE
[NFC] Minor simplifications in wasm-type.cpp

### PR DESCRIPTION
**Pull request by tlively**
Remove an unused `isFinalized` field and slightly simplify other miscellaneous
code.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5678** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5678" target="_blank">Graphite</a> 
    * **PR #5680** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5680" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5680?utm_source=stack-comment).

File: src/wasm/wasm-type.cpp [hunk 1]
```diff
@@ -822,25 +815,24 @@ bool Type::isFunction() const {
 
 bool Type::isData() const {
   if (isBasic()) {
-    // The only basic type that is considered data is a string.
-    return isString();
+    return false;
```
Comment (kripken) [inline src/wasm/wasm-type.cpp:818]: This change in classification seems nontrivial. Why is it ok?

Comment (tlively) [reply src/wasm/wasm-type.cpp:818]: There are no basic types (note: not heap types) for which `isString` returns `true` 😆

Comment (kripken) [reply src/wasm/wasm-type.cpp:818]: Ah yes, types and not heap types...

**Review by kripken — COMMENTED** [review]
Why does the diff here still show the large preceding PR? Does it need to be manually rebased despite graphite?

**Comment by tlively** [comment]
Yeah, that's a GitHub limitation, unfortunately. Graphite makes the rebase really easy, but I still have to do it for GitHub to show the correct diff.

**Review by kripken — APPROVED** [review]

**Pull request merged**